### PR TITLE
fix Issue 13285 - wrong codegen for destructor call of unnamed struct…

### PR DESCRIPTION
**Pull request by rainers**
… instance on 64 bit environments

stack alignment is not coupled with -fPIC

### CODE REVIEW

**Comment by rainers** [comment]
Just stumbled across this on Win64, not sure if the patch is correct on all platforms. If it's approved I can rebase to stable.

File: src/dmd/backend/cod2.d [hunk 1]
```diff
@@ -5266,20 +5266,15 @@ void cdddtor(ref CodeBuilder cdb,elem *e,regm_t *pretregs)
...
-            {   nalign = STACKALIGN - REGSIZE;
-                cod3_stackadj(cdb, nalign);
-            }
-            calledafunc = 1;
-            genjmp(cdb,0xE8,FLcode,cast(block *)c);   // CALL Ldtor
-            if (nalign)
-                cod3_stackadj(cdb, -nalign);
+        int nalign = 0;
+        if (STACKALIGN >= 16)
+        {   nalign = STACKALIGN - REGSIZE;
```
Comment (thewilsonator) [inline src/dmd/backend/cod2.d]: fix the style please.

**Review by thewilsonator — APPROVED** [review]

**Review by WalterBright — APPROVED** [review]

File: src/dmd/backend/cod2.d (hunk 1 above)
Comment (rainers) [reply src/dmd/backend/cod2.d]: Done.

Comment (thewilsonator) [reply src/dmd/backend/cod2.d]: Rebase to stable and its good to go.

**Comment by rainers** [comment]
Rebased again. Not sure what was wrong, but it was compiling some strange mixture of master and stable on Appveyor.

**Review by PetarKirov — APPROVED** [review]

**Comment by ZombineDev** [comment]
Auto-merge toggled on

**Pull request merged**